FEATURES.md

```diff
@@ -0,0 +1,34 @@
+# Feature Maturity
+
+Maturity of features in Istio indicate to users what they can [expect](https://istio.io/latest/about/feature-stages/) by using these
+features. In Istio, we track these requirements using a [template](features/feature_template.md). Once all
+requirements for promoting a feature to the next level are met, the [features](features.yaml)
+file may be updated to indicate this maturity level.
+
+## Features.yaml
+
+Features.yaml tracks the maturity of all Istio features. This is written
+in YAML so that it can be easily processed by any tooling that needs to
+track feature maturity. Each feature in Istio should have a features
+entry as described below.
+
+```
+features:
+  - name: "Protocols:HTTP1.1/HTTP2/gRPC/TCP"
+    maturityLevel: Stable
+    state: active
+    stateDetails: 
+      nextStateChange:
+      deprecationAnnouncement:
+    nextPromotionBefore:
+    promotionChecklist:
+```
+
+* *****Name** indicates the name of the feature.
+* ********maturityLevel** indicates the maturity level of a feature. This can be `experimental`, `alpha`, `beta`, or `stable`. Experimental is an optional maturity level that can be skipped. For alpha, beta, and stable, all requirements for the previous levels must be met before a feature can meet the next level (i.e. a stable feature has met the requirements of both alpha and beta). There may be features for which a requirement is not applicable.
+  * **state** indicates the current state of a feature. This can be `active`,`deprecated`, or `retired`. Active features are safe to use according to the [feature phase definitions](https://istio.io/latest/about/feature-stages/). While a feature is deprecated, both the deprecated feature and any replacement are supported but users should move off of these deprecated feature as it will be removed in a future release. Retired features are no longer supported and should not be used. Deprecation time varies depending on the [current maturity level](https://istio.io/latest/about/feature-stages/) and may only be declared in an LTS release.
```

Comment:
@brian-avery I wonder if state should be a structure, like:

```
state{
    name string
    begin time
    details string
}
```

Then you could have:

```
feature {
    state state
    nextState state
}
```

Reply:
I like it. Need to tie in maturity as well. Maturity needs to make sure that a promotion takes place before x date to confirm that it's staying on track, while state advances automatically on a date. So something like:

feature {
  name string
  level level
  state state
  nextState state
}

level{
  checkList string
  currentLevel enum{experimental,alpha,beta,stable}
  promotionBefore date
}

state{
    state enum{active,deprecated,retired}
    begin time
    details string
}